workflow-steps/cache/hashing-utils.ts

```diff
@@ -54,6 +54,7 @@ export function buildCachePaths(inputPaths: string) {
       inputPaths
         .split('\n')
         .filter((p) => p)
+        .map((p) => p.replace(/^~/, '..'))
```

Comment:
we need to transform the tilda into a relative path - since the executor runs in `$HOME/workspace` then replacing tilda with `..` will take one level up, into the correct place